### PR TITLE
Use fail() instead of assert(false) in tests

### PR DESCRIPTION
**Pull request by zaki50**
`assert(false)` checks nothing...

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/internal/JNIQueryTest.java [hunk 1]
```diff
@@ -444,70 +444,73 @@ public void testQueryWithWrongDataType() {
...
         // Compares strings in non string columns.
         for (int i = 0; i <= 6; i++) {
-            try { query.equalTo(new long[]{i}, oneNullTable, "string");                 assert(false); } catch(IllegalArgumentException e) {}
-            try { query.notEqualTo(new long[]{i}, oneNullTable, "string");              assert(false); } catch(IllegalArgumentException e) {}
-            try { query.beginsWith(new long[]{i}, oneNullTable, "string");              assert(false); } catch(IllegalArgumentException e) {}
-            try { query.endsWith(new long[]{i}, oneNullTable, "string");                assert(false); } catch(IllegalArgumentException e) {}
-            try { query.like(new long[]{i}, oneNullTable, "string");                    assert(false); } catch(IllegalArgumentException e) {}
-            try { query.contains(new long[]{i}, oneNullTable, "string");                assert(false); } catch(IllegalArgumentException e) {}
+            if (i != 6) {
+                try { query.equalTo(new long[]{i}, oneNullTable, "string");    fail(); } catch(IllegalArgumentException ignore) {}
```
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNIQueryTest.java]: `ignore` should be `expected` to suppress the AS lint warnings.

Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNIQueryTest.java]: `expected` doesn't suppress `Empty 'catch' block` warnings on my environment (AS 3.0 Preview4).

Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNIQueryTest.java]: uh... what?? that is shocking ... then we need to change all the `expected` to `ignore`???

Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNIQueryTest.java]: Seems yes, or something wrong in my environment?

Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNIQueryTest.java]: `ignore` is fine with my AS 2.3.3 too! I must remember something wrong ... 👍

**Pull request merged**